tests/integration_tests/functional/test_cpu_template_helper.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests that verify the cpu-template-helper's behavior."""
+
+import json
+import platform
+from pathlib import Path
+
+import pytest
+
+from framework import defs, utils
+from host_tools import cargo_build
+
+PLATFORM = platform.machine()
+
+
+# pylint: disable=too-few-public-methods
+class CpuTemplateHelper:
+    """
+    Class for CPU template helper tool.
+    """
+
+    # Class constants
+    BINARY_NAME = "cpu-template-helper"
+    BINARY_PATH = Path(
+        f"{defs.FC_WORKSPACE_TARGET_DIR}/"
+        f"{cargo_build.DEFAULT_BUILD_TARGET}/"
+        f"release/{BINARY_NAME}"
+    )
+
+    def __init__(self):
+        """Build CPU template helper tool binary"""
+        if not self.BINARY_PATH.exists():
+            utils.run_cmd(f"cd {defs.FC_WORKSPACE_DIR}")
+            utils.run_cmd(
+                f"RUSTFLAGS='{cargo_build.get_rustflags()}' "
+                f"cargo build -p {self.BINARY_NAME} --release "
+                f"--target {cargo_build.DEFAULT_BUILD_TARGET}"
+            )
+            utils.run_cmd(f"strip --strip-debug {self.BINARY_PATH}")
+
+    def dump(self, vm_config_path, output_path):
+        """Dump CPU config in JSON format by calling dump subcommand"""
+        cmd = (
+            f"{self.BINARY_PATH} dump --config {vm_config_path} --output {output_path}"
+        )
+        utils.run_cmd(cmd)
+
+
+@pytest.fixture(scope="session", name="cpu_template_helper")
+def cpu_template_helper_fixture():
+    """Fixture of CPU template helper tool"""
+    return CpuTemplateHelper()
+
+
+def save_vm_config(microvm, vm_config_path):
+    """
+    Save VM config into JSON file.
+    """
+    config_json = microvm.full_cfg.get().json()
+    config_json["boot-source"]["kernel_image_path"] = str(microvm.kernel_file)
+    config_json["drives"][0]["path_on_host"] = str(microvm.rootfs_file)
+    with open(vm_config_path, "w", encoding="utf-8") as ofile:
+        ofile.write(json.dumps(config_json, indent=4))
```

Comment:
nit: `Path(vm_config_path).write_file(json.dumps(config_json), encoding="utf-8")`

Not sure we need the UTF-8, but we probably use it everywhere in the integ tests anyway.

Reply:
Looks better! Thanks! For "utf-8", the editor warns me, but I agree that we can remove it.